javascript/Spread_operator.md

```diff
@@ -0,0 +1,301 @@
+
+#### Using Splice to Remove Array Elements in JavaScript
+The splice method can be used to add or remove elements from an array. The first argument specifies the location at which to begin adding or removing elements. The second argument specifies the number of elements to remove. The third and subsequent arguments are optional; they specify elements to be added to the array.
+
+Here we use the splice method to remove two elements starting from position three (zero based index):
+```
+var arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 0];var removed = arr.splice(2,2);/*removed === [3, 4]arr === [1, 2, 5, 6, 7, 8, 9, 0]*/
+```
+An array containing the removed elements is returned by the splice method. You can see the removed array contains [3, 4] and the original array contains the remaining values.
+
+The splice method can also be used to remove a range of elements from an array.
+```
+["bar", "baz", "foo", "qux"]list.splice(0, 2) // Starting at index position 0, remove two elements ["bar", "baz"] and retains ["foo", "qux"].
+```
+
+
+### The Slice method takes 2 arguments.
+
+**1st Argument:** Specifies from where the selection should be started.
+
+For Example:
+
+```
+var arr1 = [1,5,8,9];
+arr1.slice(1); // [5,8,9]
+From the first index (5) it will return the elements.
+```
+
+**2nd Argument:** Specifies at which level the endpoint should be. If you didn’t put this in the parenthesis while calling the slice method, it will return the elements from the starting index to the end of the array.
+
+```
+var arr1 = [1,5,8,9];
+console.log(arr1.slice(1,3));
+//[ 5, 8 ]
+```
+
+If you put a negative number while calling, the selection will be selected from the end of the array.
+
+```
+var arr1 = [1,5,8,9];
+console.log(arr1.slice(-2));
+//[ 8, 9 ]
+```
+
+Note: Slice always returns the selected elements from the array.
+
+Slice won’t change the array. The array remains intact. See the below Example:
+
+```
+var arr1 = [1,5,8,9];
+arr1.slice(2);
+console.log(arr1);
+// [ 1, 5, 8, 9 ]
+```
+
+Even if you made some changes to the array it won’t affect it. It will return the original array as it is at the start.
+[From freecodecamp](https://www.freecodecamp.org/news/lets-explore-slice-splice-spread-syntax-in-javascript-e242a6f21e60/)
+
+**Example:**
+```
+var myArray = ['a', 'b', 'c'];
+var copyOfMyArray = myArray;
+copyOfMyArray.splice(0, 1);
+alert(myArray); // alerts ['b','c']
+alert(copyOfMyArray); // alerts ['b','c']
+
+var myNumber = 5;
+var copyOfMyNumber = myNumber;
+copyOfMyNumber = copyOfMyNumber - 1;
+alert(myNumber); // alerts 5
+alert(copyOfMyNumber); // alerts 4 
+```       
+This code declares a variable `myArray` and sets it to an array value. It then declares a second variable `copyOfMyArray` and sets it to `myArray`. It performs an operation on `copyOfMyArray` and then alerts both `myArray` and `copyOfMyArray`. Somehow, when I perform an operation on `copyOfMyArray`, it appears that the same operation is performed on `myArray`.
```

Comment:
@urbashi123 You shouldn't write `Somehow`, you need to properly state the reasons why the original array `myArray` also gets mutated on updating the `copyofmyarray`.

Reply:
@urbashi123 Please address this.